lsp-types/src/Language/LSP/Types/DocumentSymbol.hs

```diff
@@ -232,7 +232,7 @@ data SymbolInformation =
     , _kind          :: SymbolKind -- ^ The kind of this symbol.
     , _tags          :: Maybe (List SymbolTag) -- ^ Tags for this symbol.
     , _deprecated    :: Maybe Bool -- ^ Indicates if this symbol is deprecated. Deprecated, use tags instead.
-    -- | The location of this symbol. The location's range is used by a tool
+    , -- | The location of this symbol. The location's range is used by a tool
```

Comment:
Fixes misplaced comment

![Screenshot from 2023-01-26 13-23-11](https://user-images.githubusercontent.com/2716069/214836579-3ea26883-b410-4c65-91ea-24ed89645bcf.png)


Mixing `-- ^` and `-- |` style comments in records leads to misplaced comments.
In these cases `-- |` is not interpreted as beginning of the next comment, but as a continuation of -- ^ comment.

This issue is relatively pervasive in this package and I can think of 2 approaches to fixing these:

1. move commas to make it clear to haddock that this comment belongs to the next field 
i.e. from
```
-- | bla
, somefield :: Int
```
to
```
, -- | bla
somefield :: Int
```

3. switch all `-- ^` comments within records to `-- |` style
This would mean more lines of code changed, but also more consistency.

Would you prefer 1 or 2?

Reply:
I think option 1 is a bit weird, maybe we switch it to 2? I personally prefer '-- |` comments.

Or as a compromise, what about this: any comment which contains at least one `-- |` comment must contain *all* `-- |` comments. And the ones that are purely `-- ^` can be left as is.